content/modules/flow-ui/pages/views/views.adoc

```diff
@@ -0,0 +1,402 @@
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=attach-event]
+----
+
+NOTE: An event is fired by Vaadin, since a view is a UI component too.
+
+[[before-close-event]]
+=== BeforeCloseEvent
+
+`BeforeCloseEvent` is the first event in the view closing process. The view is still displayed and fully functional. In this event listener, you can check any conditions and prevent closing using the `preventClose()` method of the event.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=before-close-event]
+----
+
+NOTE: In case of navigation, you need to postpone it too. Check the type of action and if it's `NavigateCloseAction`, then obtain Vaadin's `BeforeLeaveEvent` and call its `postpone()` method. A more complex example that considering both *dialog* mode and *navigation* can be found in the `io.jmix.flowui.view.StandardDetailView#preventUnsavedChanges` method which handles the default `preventUnsavedChanges` logic.
+
+
+[[after-close-event]]
+=== AfterCloseEvent
+
+`AfterCloseEvent` is the second event (after `View.BeforeCloseEvent`) in the view closing process. In this event listener, you can show notifications or dialogs after closing the view, for example:
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=notifications-bean;after-close-event]
+----
+
+[[detach-event]]
+=== DetachEvent
+
+`DetachEvent` is fired before a view is detached from the UI. This event listener can be used for releasing resources acquired by the view.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=detach-event]
+----
+
+NOTE: An event is fired by Vaadin, since a view is a UI component too.
+
+[[query-parameters-change-event]]
+=== QueryParametersChangeEvent
+
+`QueryParametersChangeEvent` is an event that informs which query parameters a view was opened with.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=query-parameters-change-event]
+----
+
+[[init-entity-event]]
+=== InitEntityEvent
+
+`InitEntityEvent` is fired in views inherited from `StandardDetailView` before the new entity instance is set to edited entity container.
+
+Use this event listener to initialize default values in the new entity instance, for example:
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserDetailEventsView.java[tags=init-entity-event]
+----
+
+[[validation-event]]
+=== ValidationEvent
+
+`ValidationEvent` is fired in views inherited from `StandardDetailView` when the view is validated on saving the view data context. Use this event listener to perform additional validation of the view.
```

Comment:
"... on saving the view's xref:...[DataContext]."
Add links to DataContext for the first occurance in each subsection. For further mentions, if any, use `DataContext`.